lib/Optimizer/Dialect/Quake/QuakeOps.cpp

```diff
@@ -895,7 +895,8 @@ void quake::U3Op::getOperatorMatrix(Matrix &matrix) {
     phi *= -1;
     lambda *= -1;
   }
-
+  // ASKME: Is this column-major representation? Or is that in
+  // `runtime/nvqir/NVQIR.cpp` - `__quantum__qis__u3` wrong?
```

Comment:
This is indeed a column-major representation, while the one in `runtime/nvqir/NVQIR.cpp` is a row-major one. Matrices must be in row-major! (Hence this one, is wrong.)

Reply:
Looking at other operations in this file, it seems to me that they all are column-major, for e.g. https://github.com/NVIDIA/cuda-quantum/blob/main/lib/Optimizer/Dialect/Quake/QuakeOps.cpp#L909